src/fs/filter.rs

```diff
@@ -288,14 +288,24 @@ impl SortField {
                 order            => order,
             },
 
-            Self::Extension(ABCabc) => match a.ext.cmp(&b.ext) {
-                Ordering::Equal  => natord::compare(&a.name, &b.name),
-                order            => order,
+            Self::Extension(ABCabc) => {
+                // Ignore extensions for directories when sorting.
+                let left = if a.is_directory() { &None } else { &a.ext };
+                let right = if b.is_directory() { &None } else { &b.ext };
+                match left.cmp(right) {
+                    Ordering::Equal  => natord::compare(&a.name, &b.name),
+                    order            => order,
+                }
             },
 
-            Self::Extension(AaBbCc) => match a.ext.cmp(&b.ext) {
-                Ordering::Equal  => natord::compare_ignore_case(&a.name, &b.name),
-                order            => order,
+            Self::Extension(AaBbCc) => {
+                // Ignore extensions for directories when sorting.
+                let left = if a.is_directory() { &None } else { &a.ext };
+                let right = if b.is_directory() { &None } else { &b.ext };
+                match left.cmp(right) {
+                    Ordering::Equal  => natord::compare_ignore_case(&a.name, &b.name),
+                    order            => order,
+                }
             },
```

Comment:
I would prefer if both code paths were combined like this:

```rust
            Self::Extension(case) => {
                // Ignore extensions for directories when sorting.
                let left = if a.is_directory() { &None } else { &a.ext };
                let right = if b.is_directory() { &None } else { &b.ext };
                match (left.cmp(right), case){
                    (Ordering::Equal, AaBbCc)  => natord::compare(&a.name, &b.name),
                    (Ordering::Equal, AaBbCc)  => natord::compare_ignore_case(&a.name, &b.name),
                    (order, _) => order,
                }
            },
```

Reply:
Thanks, done!